src/test/java/com/timgroup/statsd/NonBlockingStatsDClientMaxPerfTest.java

```diff
@@ -41,43 +41,67 @@
     @Parameters
     public static Collection<Object[]> data() {
         return Arrays.asList(new Object[][] {
-                 { 30, 17255, 256, 1, 1 },  // 30 seconds, 17255 port, 256 qSize, 1 worker
-                 { 30, 17255, 512, 1, 1 },  // 30 seconds, 17255 port, 512 qSize, 1 worker
-                 { 30, 17255, 1024, 1, 1 },  // 30 seconds, 17255 port, 1024 qSize, 1 worker
-                 { 30, 17255, 2048, 1, 1 },  // 30 seconds, 17255 port, 2048 qSize, 1 worker
-                 { 30, 17255, 4096, 1, 1 },  // 30 seconds, 17255 port, 4096 qSize, 1 worker
-                 // { 30, 17260, Integer.MAX_VALUE, 1 },  // 30 seconds, 17255 port, MAX_VALUE qSize, 1 worker
-                 { 30, 17255, 256, 2, 1 },  // 30 seconds, 17255 port, 256 qSize, 2 workers
-                 { 30, 17255, 512, 2, 1 },  // 30 seconds, 17255 port, 512 qSize, 2 workers
-                 { 30, 17255, 1024, 2, 1 },  // 30 seconds, 17255 port, 1024 qSize, 2 workers
-                 { 30, 17255, 2048, 2, 1 },  // 30 seconds, 17255 port, 2048 qSize, 2 workers
-                 { 30, 17255, 4096, 2, 1 },  // 30 seconds, 17255 port, 4096 qSize, 2 workers
-                 // // { 30, 17255, Integer.MAX_VALUE, 2 }  // 30 seconds, 17255 port, MAX_VALUE qSize, 2 workers
-                 { 30, 17255, 256, 1, 2},  // 30 seconds, 17255 port, 256 qSize, 1 sender worker, 2 processor workers
-                 { 30, 17255, 512, 1, 2 },  // 30 seconds, 17255 port, 512 qSize, 1 sender worker, 2 processor workers
-                 { 30, 17255, 1024, 1, 2 },  // 30 seconds, 17255 port, 1024 qSize, 1 sender worker, 2 processor workers
-                 { 30, 17255, 2048, 1, 2 },  // 30 seconds, 17255 port, 2048 qSize, 1 sender worke, 2 processor workers
-                 { 30, 17255, 4096, 1, 2 },  // 30 seconds, 17255 port, 4096 qSize, 1 sender worke, 2 processor workers
-                 // // { 30, 17255, Integer.MAX_VALUE, 1, 2 },  // 30 seconds, 17255 port, MAX_VALUE qSize, 1 worker
-                 { 30, 17255, 256, 2, 2 },  // 30 seconds, 17255 port, 256 qSize, 2 sender workers, 2 processor workers
-                 { 30, 17255, 512, 2, 2 },  // 30 seconds, 17255 port, 512 qSize, 2 sender workers, 2 processor workers
-                 { 30, 17255, 1024, 2, 2 },  // 30 seconds, 17255 port, 1024 qSize, 2 sender workers, 2 processor workers
-                 { 30, 17255, 2048, 2, 2 },  // 30 seconds, 17255 port, 2048 qSize, 2 sender workers, 2 processor workers
-                 { 30, 17255, 4096, 2, 2 }  // 30 seconds, 17255 port, 4096 qSize, 2 sender workers, 2 processor workers
-                 // // { 30, 17255, Integer.MAX_VALUE, 2 }  // 30 seconds, 17255 port, MAX_VALUE qSize, 2 sender workers
+                 { 30, false, 256, 1, 1 },  // 30 seconds, non-blocking, 256 qSize, 1 worker
```

Comment:
Note: This collection could be created with 3 "for loop" (non-blocking, Qsize, worker count)